catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java

```diff
@@ -231,6 +242,31 @@ public final T getByName(UriInfo uriInfo, String fqn, Fields fields) throws IOEx
     return getResultList(entities, beforeCursor, afterCursor, total);
   }
 
+  @Transaction
+  public final ResultList<T> listAfter(
+      UriInfo uriInfo, Fields fields, String fqnPrefix, int limitParam, String after, Include include)
+      throws GeneralSecurityException, IOException, ParseException {
+    // forward scrolling, if after == null then first page is being asked
+    List<String> jsons =
+        dao.listAfter(fqnPrefix, limitParam + 1, after == null ? "" : CipherText.instance().decrypt(after), include);
+
+    List<T> entities = new ArrayList<>();
+    for (String json : jsons) {
+      T entity = withHref(uriInfo, setFields(JsonUtils.readValue(json, entityClass), fields));
+      entities.add(entity);
+    }
+    int total = dao.listCount(fqnPrefix, include);
+
+    String beforeCursor;
+    String afterCursor = null;
+    beforeCursor = after == null ? null : getFullyQualifiedName(entities.get(0));
+    if (entities.size() > limitParam) { // If extra result exists, then next page exists - return after cursor
+      entities.remove(limitParam);
+      afterCursor = getFullyQualifiedName(entities.get(limitParam - 1));
+    }
+    return getResultList(entities, beforeCursor, afterCursor, total);
+  }
+
```

Comment:
We should just have one method. Include can have the default value of not-deleted when the query param is not passed.